java-lib/src/main/java/com/wavefront/ingester/AbstractIngesterFormatter.java

```diff
@@ -190,19 +201,85 @@ public void consume(StringParser parser, T target) {
     }
   }
 
+  /**
+   * Optimize the means/counts pair if necessary .
+   */
+  public static void optimizeForStorage(@Nullable List<Double> means,
+                                        @Nullable List<Integer> counts,
+                                        int size, int storageAccuracy) {
+    if (means == null || means.isEmpty() || counts == null || counts.isEmpty()) {
+      return;
+    }
+
+    if /*Too many centroids*/ (size > 5 * storageAccuracy) {
```

Comment:
we might want to reduce this `5` times of accuracy limit here. since `5 * 32 = 160` which is larger then our default 100 centroids limit. I suggest we do `2` here.

Reply:
Sure, sounds good to me